server/src/main/java/org/opensearch/index/shard/RemoteSegmentGarbageCollector.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.shard;
+
+public interface RemoteSegmentGarbageCollector {
+
+    interface Listener {
+
+        /**
+         * Called after the successful deletion of stale segments from remote store
+         */
+        void onSuccess(Void response);
+
+        /**
+         * Called on exception while deleting stale segments from remote store
+         */
+        void onFailure(Exception e);
+
+        /**
+         * Called on skipping the garbage collection
+         */
+        void onSkip(String reason);
```

Comment:
Maybe a better contract would be shouldGarbageCollect() that returns a boolean. We can add additional safety constraints explicitly